verilog/analysis/verilog_project.cc

```diff
@@ -185,6 +186,25 @@ bool VerilogProject::RemoveRegisteredFile(
   return false;
 }
 
+std::string VerilogProject::GetRelativePathToSource(
+    const absl::string_view absolute_filepath) {
+  // TODO add check if the absolute_filepath is out of the VerilogProject
+  std::filesystem::path absolutepath{
+      static_cast<std::string>(absolute_filepath)};
+  std::filesystem::path projectpath{std::string{TranslationUnitRoot()}};
+  auto relative = std::filesystem::relative(absolutepath, projectpath);
+  return relative.string();
+}
+
+absl::Status VerilogProject::updateFileContents(
+    absl::string_view path, const verible::TextStructureView* updatedtext) {
+  auto projectpath = GetRelativePathToSource(path);
+  files_[projectpath] =
```

Comment:
Instead of the `operator []` we usually prefer to explicitly call `insert()`, as it also allows from the return value to deduce if we had to replace a content (which might or might not be desired, but we have a chance to react to it).